engine/storage/volume/src/main/java/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -468,6 +471,7 @@ public Void deleteVolumeCallback(AsyncCallbackDispatcher<VolumeServiceImpl, Comm
                         _snapshotStoreDao.remove(snapStoreVo.getId());
```

Comment:
Hi @weizhouapache, isn't it better to keep the info in the DB for the snapshots and update the State as `Destroyed`?

Reply:
@slavkap 
is there any benifit ?
the records will be cleaned up by storage garbage collector.

Reply:
@slavkap 
thanks for testing

Reply:
@weizhouapache, to keep it as history, nothing more

Reply:
@slavkap ah ok, thanks for reply.
the records will be finally cleaned up by storage garbage collector (by default run once every day). They will be kept for only 1 day.
The behaviour is not changed by this PR. It was introduced by 336df84f1787de962a67d0a34551f9027303040e and ccb68e13a74342d4798aae11fe5c57890ff6c0ae